drivers/volume/kdmp/kdmp.go

```diff
@@ -749,12 +751,22 @@ func (k *kdmp) StartRestore(
 			pvc.Name = bkpvInfo.PersistentVolumeClaim
 			pvc.Namespace = restoreNamespace
 		}
-		// Check if the snapshot size is larger than the pvc size.
-		// Update the pvc size to that of the snapshot size if so.
+		// Below two cases, we will change the PVC original size.
+		// case1: If the snapshot size is greater than the PVC size. In some of the filesystem
+		// provisioned PVC, the content of the PVC is allowed to grow beyond the PVC size.
+		// case2: If the snapshot size + 10% of the snapshot size goes beyind the PVC size.
+		// We also allow the customer to configure the sizePercentage to configured, if the 10% does not
+		// satisfy their storage provisioner.
+		// KDMP_RESTORE_PVC_SIZE_PERCENTAGE parameter need to be added to the kdmp-config configmap in kube-system namespace.
+		sizePercentage := utils.GetRestorePvcSizePercentage()
+		if sizePercentage == 0 {
+			sizePercentage = defaultPvcSizeIncreasePercentage
+		}
+		logrus.Infof("StartRestore: sizePercentage is - %v", sizePercentage)
+		newPvcQuantity := resource.NewQuantity(int64(bkpvInfo.TotalSize)+((int64(sizePercentage)*int64(bkpvInfo.TotalSize))/int64(100)), resource.BinarySI)
```

Comment:
nit : For readability
```suggestion
		totalSize := int64(bkpvInfo.TotalSize)
		newPvcQuantity := resource.NewQuantity(
			totalSize+(totalSize*(int64(sizePercentage))/int64(100)),
			resource.BinarySI,
		)
```

Reply:
Fixed